lib/spack/spack/reporters/cdash.py

```diff
@@ -297,6 +307,9 @@ def upload(self, filename):
             request = Request(url, data=f)
             request.add_header('Content-Type', 'text/xml')
             request.add_header('Content-Length', os.path.getsize(filename))
+            if self.authtoken:
+                request.add_header('Authorization',
+                                   'Bearer {0}'.format(self.authtoken))
```

Comment:
Can the same authorization token be used both here and where we create/populate a build group?

Reply:
Yes. Generally speaking CDash's authentication tokens allow you to take any action that the user who created them is authorized to perform.